api/src/main/java/org/pmiops/workbench/api/OfflineWorkspaceController.java

```diff
@@ -0,0 +1,29 @@
+package org.pmiops.workbench.api;
+
+import java.util.List;
+import org.pmiops.workbench.cloudtasks.TaskQueueService;
+import org.pmiops.workbench.workspaces.WorkspaceService;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.RestController;
+
+@RestController
+public class OfflineWorkspaceController implements OfflineWorkspaceApiDelegate {
+  private final TaskQueueService taskQueueService;
+  private final WorkspaceService workspaceService;
+
+  @Autowired
+  public OfflineWorkspaceController(
+      TaskQueueService taskQueueService, WorkspaceService workspaceService) {
+    this.taskQueueService = taskQueueService;
+    this.workspaceService = workspaceService;
+  }
+
+  @Override
+  public ResponseEntity<Void> cleanupOrphanedWorkspaces() {
```

Comment:
Just a gut-check... how do we express security elements on this endpoint and the batch one above to require a service account auth and not just any requestor? (prevent DoS attack).